x/crosschain/types/cctx.go

```diff
@@ -203,6 +205,51 @@ func (m CrossChainTx) SetOutboundBallotIndex(index string) {
 }
 
 // GetCctxIndexFromBytes returns the CCTX index from a byte array.
+func (m CrossChainTx) GetValidReceiverAddress() (ethcommon.Address, error) {
+	inboundCoinType := m.InboundParams.CoinType
+	switch inboundCoinType {
+	// For coin type ZETA the receiver is added to the receiver field of the inbound vote so we can use it directly
+	case coin.CoinType_Zeta:
+		{
+			err := address.ValidateEthereumAddress(m.GetCurrentOutboundParam().Receiver)
+			if err != nil {
+				return ethcommon.Address{}, cosmoserrors.Wrap(ErrInvalidReceiverAddress, err.Error())
+			}
+			return ethcommon.HexToAddress(m.GetCurrentOutboundParam().Receiver), nil
+		}
+	// For coin type gas we create the message by composing the receiver and the call data so we need to parse the address from the RelayedMessage
+	// The receiver in this case is set to me the sender of the message
+	// For cases where this is not the case we should try and use the receiver field of the inbound vote
```

Comment:
This logic will be removed for V2 contracts, receiver will always contain the receiver address (which should have always been the case)